### PR TITLE
[batch] added async_os and escaping helper functions

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: hail/python/hailtop/utils/os.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    shutil.copy(src, dest)
+
+
+def remove(path):
+    if os.path.exists(path):
+        assert os.path.isfile(path)
+        os.remove(path)
+
+
+def new_file(path, size):
```
Comment (danking) [inline hail/python/hailtop/utils/os.py:90]: let's call this `touch` with a default size of 0.

File: hail/python/hailtop/utils/os.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    while i < n:
+        if i < n - 1 and c[i] == '\\' and c[i + 1] in wildcards:
+            new_c.append(c[i + 1])
+            i += 2
+            continue
+        elif c[i] in wildcards:
+            return ''.join(new_c)
+        new_c.append(c[i])
+        i += 1
+    return ''.join(new_c)
```
Comment (danking) [inline hail/python/hailtop/utils/os.py:71]: Can we leave these four functions out until we need them for something like `glob`?

**Comment by danking** [comment]
Also, does `import hailtop.utils.os as hos` work or should we add `os` to the all list for utils?

**Comment by jigold** [comment]
> Also, does import hailtop.utils.os as hos work or should we add os to the all list for utils?

This works

**Comment by jigold** [comment]
Stacked on #8610

**Comment by jigold** [comment]
This doesn't need to be stacked.

**Comment by jigold** [comment]
I think this should go somewhere else.

**Pull request closed without merging**